Lib/zipapp.py

```diff
@@ -131,14 +131,38 @@ def create_archive(source, target=None, interpreter=None, main=None,
     elif not hasattr(target, 'write'):
         target = pathlib.Path(target)
 
+    # Create the list of files to add to the archive now, in case
+    # the target is being created in the source directory - we
+    # don't want the target being added to itself
+    files_to_add = sorted(source.rglob('*'))
+
+    # The target cannot be in the list of files to add. If it were, we'd
+    # end up overwriting the source file and writing the archive into
+    # itself, which is an error. We therefore check for that case and
+    # provide a helpful message for the user.
+
+    # Note that we only do a simple path equality check. This won't
+    # catch every case, but it will catch the common case where the
+    # source is the CWD and the target is a file in the CWD. More
+    # thorough checks don't provide enough value to justify the extra
+    # cost.
+
+    # https://github.com/python/cpython/issues/104527 tracks making
+    # the zipfile module catch writing an archive to itself at a
+    # lower level, which could help here in cases that our check
+    # doesn't catch.
+    if target in files_to_add:
```

Comment:
`target` can be a file-like object. `target in files_to_add` can invoke comparison of a file-like object with the `Path` objects. Is it what we need?

Reply:
I believe it's fine. A file-like object will simply test as not equal to all the `Path` objects. I guess I *could* add some sort of exception handling in case the user passes an object with a custom equality check that fails, but that seems like overkill to me.

I was going to add a note to that effect in the comment, but it's long enough already and I wasn't sure it was worth it. The fact that you pointed it out suggests that it would be, though.